addons/sale_service/models/procrument_order.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models, _
+
+
+class ProcurementOrder(models.Model):
+    _inherit = "procurement.order"
+
+    task_id = fields.Many2one('project.task', string='Task', copy=False)
+
+    @api.multi
+    def _is_procurement_task(self):
+        self.ensure_one()
+        return self.product_id.type == 'service' and self.product_id.auto_create_task or False
+
+    @api.v7
+    def _assign(self, cr, uid, procurement, context=None):
```

Comment:
there is need of writing two method because calling v8 from v7 and v8 is again calling super(). so need to pass self as argument.Instead make 2 method.